pkg/controller/tenant/tenant_controller.go

```diff
@@ -115,10 +125,13 @@ func NewTenantController(
 		},
 		resyncPeriod,
```

Comment:
I am really worried about using AddEventHandlerWithResyncPeriod and resyncPeriod as 5 min. This meaning every five min all tenant creation events will be reevaluated again. When we have more than ten thousand tenants, actively going over all tenants and re-evaluate will consume a lot of resources.

Reply:
the resync period is for actions like tenant deletion to work with a periodic check.